File: backend/app/models/workflow.py
import uuid

from sqlalchemy import ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base
from app.models.mixins import TimestampMixin, UUIDPrimaryKeyMixin


class Workflow(UUIDPrimaryKeyMixin, TimestampMixin, Base):
    __tablename__ = "workflows"

    name: Mapped[str] = mapped_column(String(160), index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    trigger_type: Mapped[str] = mapped_column(String(50), index=True)
    status: Mapped[str] = mapped_column(String(50), default="pending", index=True)
    created_by: Mapped[uuid.UUID | None] = mapped_column(ForeignKey("users.id"), nullable=True)

    creator: Mapped["User | None"] = relationship(back_populates="workflows")
    steps: Mapped[list["WorkflowStep"]] = relationship(back_populates="workflow", cascade="all, delete-orphan")
    executions: Mapped[list["Execution"]] = relationship(back_populates="workflow", cascade="all, delete-orphan")

File: frontend/types/workflow.ts
export type WorkflowStatus = "pending" | "running" | "completed" | "failed" | "retrying";

export type Workflow = {
  id: string;
  name: string;
  description: string;
  triggerType: string;
  status: WorkflowStatus;
  executions: number;
  successRate: number;
};

export type ExecutionLog = {
  id: string;
  workflow: string;
  status: WorkflowStatus;
  timestamp: string;
  message: string;
};

